packages/rich-text/src/component/use-anchor.js

```diff
@@ -31,6 +31,13 @@ function getFormatElement( range, editableContentElement, tagName, className ) {
 	if ( element === editableContentElement ) return;
 	if ( ! editableContentElement.contains( element ) ) return;
 
+	// Check if the candidate element is within the selected range.
+	// This avoids situations where two format elements can exist within the
+	// same parent element and the wrong element is mistakenly returned.
+	if ( ! range.isPointInRange( element, 0 ) ) {
```

Comment:
It's unclear to me what this is doing. Does this check if the point (parentElement of startContainer at index 0) is within the range? This may be a bit too restrictive? There may be cases where that element _contains_ the range, but the range does not contain the whole element, and neither the element at that point (0).